test/publicresolver_test.js

```diff
@@ -1,85 +1,384 @@
 var assert = require('assert');
 var async = require('async');
+var Promise = require('bluebird');
 
 var utils = require('./utils.js');
+Promise.promisifyAll(utils);
 var web3 = utils.web3;
+Promise.promisifyAll(web3.eth);
 
 var accounts = null;
 
-before(function(done) {
-	web3.eth.getAccounts(function(err, acct) {
-		accounts = acct
-		done();
-	});
+before(function() {
+	return web3.eth.getAccountsAsync()
+		.then(acct => accounts = acct);
 });
 
 describe('PublicResolver', function() {
 	var resolverCode = null;
 	var resolver = null;
 	var ens = null;
 
-
 	before(function() {
 		this.timeout(10000);
 		resolverCode = utils.compileContract(['interface.sol', 'PublicResolver.sol']).contracts['PublicResolver.sol:PublicResolver'];
 	});
 
-	beforeEach(function(done) {
-		async.series([
-			function(done) { ens = utils.deployENS(accounts[0], done); },
-			function(done) {
-				resolver = web3.eth.contract(JSON.parse(resolverCode.interface)).new(
-				    ens.address,
-				    {
-				    	from: accounts[0],
-				     	data: resolverCode.bytecode,
-				     	gas: 4700000
-				   	}, function(err, contract) {
-				   	    assert.equal(err, null, err);
-				   	    if(contract.address != undefined) {
-				   	    	ens.setSubnodeOwner(0, web3.sha3('eth'), accounts[0], {from: accounts[0]}, done);
-					   	}
-				   });
-			}],
-			done
-		);
-	});
+	beforeEach(function() {
+		this.timeout(4000);
+		return utils.deployENSAsync(accounts[0])
+			.then(_ens => {
+				ens = _ens;
+				return new Promise(function(resolve, reject) {
+					web3.eth.contract(JSON.parse(resolverCode.interface)).new(
+						ens.address,
+						{
+							from: accounts[0],
+							data: resolverCode.bytecode,
+							gas: 4700000
+						},
+						function(err, contract) {
+							if (err) {
+								reject(err);
+							} else if (typeof contract.address !== "undefined") {
+								resolve(contract);
+							} else {
+								// There is to hope that reject or resolve is called
+							}
+						});
 
-	it('permits setting addresses', function(done) {
-		async.series([
-			function(done) { resolver.supportsInterface("0x3b3b57de", function(err, result) {
-					assert.equal(err, null, err);
-					assert.equal(result, true);
-					done();
-				});
-			},
-			function(done) { resolver.setAddr(utils.node, accounts[1], {from: accounts[0]}, function(err, txid) {
-					assert.equal(err, null, err);
-					done();
 				});
-			},
-			function(done) { resolver.addr(utils.node, function(err, address) {
-					assert.equal(err, null, err);
-					assert.equal(address, accounts[1]);
-					done();
+			})
+			.then(contract => {
+				resolver = Promise.promisifyAll(contract);
+				ens.setSubnodeOwnerAsync(0, web3.sha3('eth'), accounts[0], {from: accounts[0]});
+			});
+	});
+
+	describe('constructor', function() {
+
+		it('should not allow a 0 address for ens');
```

Comment:
Right, but the same goes for deploying it with an invalid address for ENS; either way, your only option is to redeploy the contract. Further, if we did have this check, it would cause the deployment to fail, leading to the exact same outcome - a moribund contract.